### ISSUE TITLE
[Feature] Enhancements to the Tags System, File Details and Settings

### ISSUE THREAD

**Issue by Conveniences67** [stashapp/stash#5273]
1. Ability to color code tags, whether it be the frame around the the tag or the actual word to help quickly differentiate tags

To Make it easier to differentiate between male/female/others actors by their names 
or what character their playing as, or in the case of the latter portion in this list of suggestions what group it falls under or sexual act is being performed
 
Maybe even a multicolor option, that's preset to help differentiate the primary sexual orientation being displayed like how gay = rainbow, but have the multicolor option be based on the currently confirmed lgbtq+ flags maybe even have it as a built in option instead of in the tag system somewhere between performers and tags

1.1.  Add to the edit an orientation drop down menu if the tag option is too tedious 

1.2 Maybe Add sex orientation on menu items

2. Option to Hide tags from the overall list and save it as a setting

Trying to make a tagging system similar to e-hentai, and currently have it set up where their's a male:, female: sub tags that's attached to the main tag in question and am hoping for an option to hide the parent tag

3. Ability to put tags into groups and able to color code those groups as well

To help make sub groups like characters for manga and doujinshi that have characters from shows/movies etc. 
Or even actions/sex acts and positions and poses

Or even manage Sub tags into groups of their own

4. Sub tags numbers being counted in the main/parent tags

5. Option to copy definition from parent tag to sub tag

6. A way to quickly get tags from the images from sites like e-hentai or Gelbooru, Pixiv, or e621, with a way to easily add more to the selection based on Metadata or the image ID that's used in the sites gallery

An option to read the image filename in cases were the Metadata or Gallery ID is in the filename 

7. Ability to start on a new tag after finishing making a new one. As well as performers 

It gets rather annoying that in order to make a new tag I have to got to tags then hit new

8. Option to have more than one Image as the set option for the gallery tag, and an additional option to have it on a delay for how long the image takes to transition

9. Option to change how the the date is display 

I perfer MM-DD-YYYY as the date format and most of my images have that written as that in the filename

Would make it easier to copy and paste in the edit portion of file details

10.  Add option to switch out photographer with artists for images that are clearly drawn, maybe even add it as a option in the menu setting

Would be nice if this had its own dedicated drop-down menu in the file details when editing along with being able to add new artist while working in the file info without it being previously added, and being able to toggle between photographer and artist

11. Parody, Franchise or Fictional Media option for situations when a fictional character is being portrayed or acted

Have it as a toggleable option in the file details when editing, and maybe as another menu item besides performers, with the default set to original or none

12. Option for written pornograpy, those who perfer the written word over visual works, option for as a menu item as well

Have an option for author, synopsis, Tags and etc.

13. Ability to change priority for the menu items, some people perfer drawn or written  pornography to videos, and would like to have the menu items reflect that

14. Ability to choose whether or not to use the aliases instead of the tag title/name

I like to use the e-hentai tagging system for my manga and doujinshi, and am hoping most for an implementation of either 2 or 3 as it's the one I'm currently using to tag my stuff, but if not could this be possible 

15. Should 9 be possible to implement, an option to color code the aliases and have them counted both separately: if that specific aliases is used, and overall: for direct relation to the tag being used

An ability to single out that specific aliases would be nice as well

Basically a combination of multiple suggestions above but focused around aliases

**Comment by echo6ix** [comment]
### Suggestion

This feature request (FR) contains multiple FRs embedded within it.  If you want to increase the odds of engagement with developers and contributors, I would suggest the following:

- Best practice is not to put a litany of FRs into one post. Create individual FRs, and cross-reference them for clarity if you need to
- Use the templates that Github auto-populates when creating a new FR. It provides a recommended layout the devs expect to see
- Before creating a FR, search Github to determine if your idea has already been requested

### Addressing Requests & Comments

Lets take a look at some of your requests. This isn't going to be a thorough examination of your post because you have a lot going on here.

> 1. Ability to color code tags, whether it be the frame around the the tag or the actual word to help quickly differentiate tags

Already exists. If you want to suggest nuance please do so here https://github.com/stashapp/stash/issues/3200

> 2. Option to Hide tags from the overall list and save it as a setting

Already exists. If you want to suggest nuance please do so here https://github.com/stashapp/stash/issues/2633

> It gets rather annoying that in order to make a new tag I have to got to tags then hit new

If I understand correctly, you can already do this. Go to `Settings` > `Interface` > `Editing` > `Disable dropdown create` > toggle switch for `Tag`

> 10. Add option to switch out photographer with artists for images that are clearly drawn, maybe even add it as a option in the menu setting

Very niche FR. A lot of people want their own arbitrary fields. Obviously Stash can't accommodate everyone. A better approach may be user-created custom fields, and this FR already exists. Add your nuance here if required https://github.com/stashapp/stash/issues/885

Alternatively, since the aforementioned FR -- #885  -- is explicitly about performer user-created fields, you could create a less specific FR for general purpose user-created custom fields extended for all Stash objects (scenes, images, galleries, groups ,etc). Be sure to cross-reference it back to #885. Search first to make sure that FR doesn't already exist.

> 12. Option for written pornograpy, those who perfer the written word over visual works, option for as a menu item as well

Already exists. This is one of the oldest FRs. https://github.com/stashapp/stash/issues/14

> 13. Ability to change priority for the menu items, some people perfer drawn or written pornography to videos, and would like to have the menu items reflect that

This concept is covered in a lot of detail by myself here on the main navbar refactor discussion https://github.com/stashapp/stash/discussions/1728

**Comment by C-BoT-AU** [comment]
> Option to change how the the date is display
I perfer MM-DD-YYYY as the date format and most of my images have that written as that in the filename

While I can't get onboard with your MM-DD-YYYY format I did make a discovery today while I was trying to address the dates myself. I had the [Qx Scene Card](https://github.com/stashapp/CommunityScripts/tree/main/plugins/qxSceneCard) extension installed, and it was formatting my dates on the cards to DD/MM/YYYY which I hate.
Given they're stored in the database as ISO formatted dates, they are being "converted" somewhere.

Unfortunately, I don't speak Javascript, but I found these lines with the plugins `qxSceneCard.js` file, and had a play around with them and by changing the timeZone, I was able to show different date formats:

```
      /* @__PURE__ */ React.createElement(
        "span",
        { className: "date" },
        date &&
          /* @__PURE__ */ React.createElement(FormattedDate, {
            value: date,
            format: "short",
            timeZone: "utc",
          })
      )
```

Unfortunately, other than these lines within the `SceneCard-######.js` file, I can't find where it is set without using the extension.
```
  , me = k("SceneCard.Details", e => u("div", {
    className: "scene-card__details",
    children: [n("span", {
        className: "scene-card__date",
        children: e.scene.date
```
